### PR TITLE
Add basic interpreter framework

### PR DESCRIPTION
**Pull request by axic**

### CODE REVIEW

File: lib/fizzy/execute.cpp [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+    {
+        auto instruction = *pc++;
+        std::cout << "Op: " << static_cast<int>(instruction) << std::endl;
+        switch (instruction)
+        {
+        case instr::end:
+            goto end;
+        case instr::local_get: {
+            auto idx = load<uint32_t>(immediates);
+            immediates += 4;
```
Comment (axic) [inline lib/fizzy/execute.cpp]: Could also change `load` to do the increment.

File: test/unittests/execute_test.cpp [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    auto ret = fizzy::execute(module, 0, {});
+
+    ASSERT_EQ(ret.size(), 1);
+    EXPECT_EQ(ret[0], 0);
+}
+
+TEST(execute, basic_add_with_inputs)
+{
+    fizzy::module module;
+    fizzy::bytes imm;
```
Comment (chfast) [inline test/unittests/execute_test.cpp]:
```suggestion
    fizzy::bytes imm{0, 0, 0, 0, 1, 0, 0, 0};
```

File: lib/fizzy/execute.cpp [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
```
Comment (chfast) [inline lib/fizzy/execute.cpp]:
```suggestion
            immediates += sizeof(uint32_t);
```

File: lib/fizzy/execute.hpp [hunk 4]
```diff
@@ -0,0 +1,10 @@
+#pragma once
+
+#include "types.hpp"
+#include <cstdint>
+
+namespace fizzy
+{
+std::vector<uint64_t> execute(
+    struct module& _module, funcidx _function, std::vector<uint64_t> _args);
```
Comment (gumb0) [inline lib/fizzy/execute.hpp]:
```suggestion
    const module& _module, funcidx _function, std::vector<uint64_t> _args);
```

**Pull request merged**